### PR TITLE
new: pool type data on v3

### LINKED ISSUE #652: sync pool specific on-chain data
need to sync pool specific on-chain data from v3 pools, such as amp for stable pools.

need to create a system that is extendable.

### PR DESCRIPTION
**Pull request by gmbronco**
closes #652

### CODE REVIEW

File: modules/actions/pool/type-data-v3/update.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { prisma } from '../../../../prisma/prisma-client';
+
+export const update = async (params: { id: string; typeData: any }[]): Promise<void> => {
```
Comment (franzns) [inline modules/actions/pool/type-data-v3/update.ts:3]: why do we need this as executeRaw?

File: modules/sources/contracts/fetch-pool-type-data.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+            switch (type) {
+                case PrismaPoolType.STABLE:
+                    return {
+                        id,
+                        typeData: parseStableContractCalls(result),
+                    };
+            }
+        })
+        .filter((x): x is { id: string; typeData: { amp: string } } => !!x);
```
Comment (franzns) [inline modules/sources/contracts/fetch-pool-type-data.ts]: Cant we just filter the undefined results?

File: modules/actions/pool/type-data-v3/sync-immutable-data-for-pools.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            if (pool.factory.type === PoolType.Stable) {
+                return {
+                    id: pool.id,
+                    typeData: {
+                        amp: pool.amp, // This is mutable and is updated in sync-dynamic-data-for-pools.ts – setting it as a temporary example
+                    },
+                };
+            }
+        })
+        .filter((x): x is { id: string; typeData: { amp: string } } => !!x);
```
Comment (franzns) [inline modules/actions/pool/type-data-v3/sync-immutable-data-for-pools.ts]: Should this just filter undefined ones?

File: modules/actions/pool/type-data-v3/update.ts (hunk 1 above)
Comment (gmbronco) [reply modules/actions/pool/type-data-v3/update.ts:3]: it's a merge vs overwrite for the JSON field. i was thinking about a case where not all of the typeData is updated by the same action. Prisma doesn't support that natively.

File: modules/sources/contracts/fetch-pool-type-data.ts (hunk 2 above)
Comment (gmbronco) [reply modules/sources/contracts/fetch-pool-type-data.ts]: yes, that's what it's doing. filtering out faulty responses. do you mean some other way to do that?

File: modules/actions/pool/type-data-v3/sync-immutable-data-for-pools.ts (hunk 3 above)
Comment (gmbronco) [reply modules/actions/pool/type-data-v3/sync-immutable-data-for-pools.ts]: yup, same as above, we expect an object, so anything falsy won't work

**Pull request merged**